addons/rook/template/base/install.sh

```diff
@@ -339,7 +339,57 @@ function rook_cluster_deploy_upgrade() {
 
     rook_cluster_deploy_upgrade_flexvolumes_to_csi
 
-    logSuccess "Rook-ceph cluster upgraded"
+    verify_rook_updated_cluster
+}
+
+# Before to finish end report that the upgrade was done with success ensure that
+# only on rook version is found and the ceph status
+# https://rook.io/docs/rook/v1.10/ceph-upgrade.html#3-verify-the-updated-cluster
+function verify_rook_updated_cluster() {
+    logStep "Checking if Rook Upgrade is completed successfully"
+    echo "Verifying Rook Version Deployed"
+    if ! spinner_until 1200 rook_version_deployed ; then
+        logWarn "Timeout awaiting check Rook version deployed"
+        logStep "Rook versions and replicas"
+        kubectl -n rook-ceph get deployment -l rook_cluster=rook-ceph -o jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \trook-version="}{.metadata.labels.rook-version}{"\n"}{end}'
+        local rook_versions=
+        rook_versions="$(kubectl -n rook-ceph get deployment -l rook_cluster=rook-ceph -o jsonpath='{range .items[*]}{"rook-version="}{.metadata.labels.rook-version}{"\n"}{end}' | sort | uniq)"
+        if [ -n "${rook_versions}" ] && [ "$(echo "${rook_versions}" | wc -l)" -gt "1" ]; then
+            logWarn "Detected multiple Rook versions"
+            logWarn "${rook_versions}"
+            bail "Failed to verify the Rook upgrade, multiple Rook versions detected"
+        fi
+    fi
+
+    echo "Verifying Ceph Version ${ceph_version} Deployed"
+    if ! spinner_until 1200 rook_ceph_version_deployed "${ceph_version}" ; then
```

Comment:
Why not use the go code here?

```suggestion
    if ! "$DIR"/bin/kurl rook wait-for-ceph-version "${ceph_version}-0" --timeout=1200 ; then
```

Reply:
Why should we add the `-0`?

Reply:
I'm not sure. I am just copying other code and in all cases we append the `-0`

Reply:
WDYT about we check it in a follow up?
What about we move forward with the shell check (which is also used in many other places) and we review this one with go to understand better the usage of `-0` and ensure that all places started to use only the go implementation and we remove the shell check if that is our desire ?

The tests after the suggestions are here: https://testgrid.kurl.sh/run/rook_upgrade_feb_6_check_upgrades_with_rook_check_with_suggestions are working fine after the sugge

Reply:
@emosbaugh ^